src/main/java/org/breedinginsight/services/writers/ExcelWriter.java

```diff
@@ -21,12 +26,21 @@ public class ExcelWriter {
     private static final int EXCEL_COLUMN_NAMES_ROW = 0;
 
     //Writes a xlsx workbook with one sheet with desired columns and data
-    public static XSSFWorkbook writeToWorkbook(String sheetName, List<Column> columns, List<Map<String, Object>> data) {
+    public static Workbook writeToWorkbook(String sheetName, List<Column> columns, List<Map<String, Object>> data, FileType extension) {
+        Workbook workbook;
+        Sheet sheet;
+
         //Create workbook
-        XSSFWorkbook workbook = new XSSFWorkbook();
+        if (extension.getName().equals("xlsx")){
+            workbook = new XSSFWorkbook();
+        } else if (extension.getName().equals("xls")) {
```

Comment:
This should use the enum for comparison, same as the comment on line 160 of the BrAPIGermplasmService

Reply:
fixed